source/_posts/2020-07-07-array-backwards-compatibility-using-property-wrappers.markdown

```diff
@@ -0,0 +1,73 @@
+---
+layout: post
+author: matt
+title: "Array backwards compatibility using Property Wrappers"
+date: 2020-07-07 14:19:05 +0200
+comments: true
+categories: 
+---
+
+Let's assume that you're doing an application which shows your users pretty pictures of seasons. You contact your backend folks and they tell you that the world is simple, there are only three seasons:
+
+```swift
+enum Season: String, Decodable {
+    case spring, summer, autumn
+}
+```
+<!--more-->
+
+In order to compose things nicely, you put the seasons into a struct:
+
+```swift
+struct Seasons: Decodable {
+    var available: [Season]
+}
+```
+
+The following JSON comes from the server, it gets decoded, everything works well:
+```json
+"available": ["spring", "summer", "autumn"]
+// [__lldb_expr_12.Season.spring, __lldb_expr_12.Season.summer, __lldb_expr_12.Season.autumn]
+```
+
+The application gets released. Time passes, winter comes and the app gets updated seasons from the backend:
+```json
+"available": ["spring", "summer", "autumn", "winter"]
+```
+
+What happens now? your functionality breaks. Since the app can't understand `Season.winter`, no seasons get decoded. You recive a lot of bug reports, and your users are not happy ☹️
```

Comment:
`Your`
Maybe a line break on `Since the` would make it look better?
typo: `receive`